server/src/main/java/org/opensearch/throttling/admissioncontrol/AdmissionControlService.java

```diff
@@ -0,0 +1,154 @@
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.throttling.admissioncontrol.controllers.AdmissionController;
+import org.opensearch.throttling.admissioncontrol.controllers.IOBasedAdmissionController;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.ConcurrentMap;
+
+import static org.opensearch.throttling.admissioncontrol.AdmissionControlSettings.IO_BASED_ADMISSION_CONTROLLER;
+
+/**
+ * Admission control Service that bootstraps and manages all the Admission Controllers in OpenSearch.
+ */
+public class AdmissionControlService {
+    private final ThreadPool threadPool;
+    public final AdmissionControlSettings admissionControlSettings;
+    private final ConcurrentMap<String, AdmissionController> ADMISSION_CONTROLLERS;
+    private static AdmissionControlService admissionControlService = null;
+    private static final Logger logger = LogManager.getLogger(AdmissionControlService.class);
+    private final ClusterSettings clusterSettings;
+    private final Settings settings;
+
+    /**
+     *
+     * @param settings Immutable settings instance
+     * @param clusterSettings ClusterSettings Instance
+     * @param threadPool ThreadPool Instance
+     */
+    public AdmissionControlService(Settings settings, ClusterSettings clusterSettings, ThreadPool threadPool) {
+        this.threadPool = threadPool;
+        this.admissionControlSettings = new AdmissionControlSettings(clusterSettings, settings, this);
+        this.ADMISSION_CONTROLLERS = new ConcurrentHashMap<>();
+        this.clusterSettings = clusterSettings;
+        this.settings = settings;
+        this.initialise();
+    }
+
+    /**
+     * Initialise and Register all the admissionControllers
+     */
+    public void initialise() {
+        // Initialise different type of admission controllers
+        registerAdmissionController(IO_BASED_ADMISSION_CONTROLLER);
+    }
+
+    /**
+     * Handler to trigger registered admissionController
+     */
+    public boolean applyTransportAdmissionControl(String action) {
+        if (this.isTransportLayerAdmissionControllerEnabled() == false && this.admissionControlSettings.isForceDefaultSettingsEnabled()) {
+            return false;
+        }
+        this.ADMISSION_CONTROLLERS.forEach((name, admissionController) -> { admissionController.acquire(action); });
+        return true;
+    }
+
+    /**
+     *
+     * @return singleton admissionControlService Instance
+     */
+    public static AdmissionControlService getInstance() {
+        return admissionControlService;
+    }
+
+    /**
+     *
+     * @param settings Immutable settings instance
+     * @param clusterSettings ClusterSettings Instance
+     * @param threadPool ThreadPool Instance
+     * @return singleton admissionControlService Instance
+     */
+    public static synchronized AdmissionControlService newAdmissionControllerService(
```

Comment:
nit: newAdmissionControlService - can you please check through all files once more to make sure naming is consistent